### PR TITLE
Fixes lasgun underbarrel attachments

### LINKED ISSUE #1078: Masterkey On M37 Bug
## Testmerges:

No testmerges. Code from 3/13. If this has been noticed and fixed since then, feel free to delete. 

## Reproduction:

Right so, just put a masterkey shotgun on an M37, and try to reload the masterkey. It automatically switches to loading the M37's tube instead. Using buckshot, which is the ammo the masterkey takes.

### PR DESCRIPTION
**Pull request by DominikPanic**
Fixes #1078

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 1]
```diff
@@ -289,21 +289,11 @@ should be alright.
...
-					active_attachable.activate_attachment(src, null, TRUE)
-					reload(user,MG)
-					return
-			active_attachable.reload_attachment(I, user)
-
-	else if(istype(I,/obj/item/ammo_magazine))
-		if(check_inactive_hand(user))
-			reload(user,I)
+	else if(active_attachable?.flags_attach_features & ATTACH_RELOADABLE && check_inactive_hand(user))
+		active_attachable.reload_attachment(I, user)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: shouldnt this return?

File: code/modules/projectiles/updated_projectiles/guns/energy.dm [hunk 2]
```diff
@@ -290,31 +290,14 @@
...
-				playsound(user, active_attachable.activation_sound, 15, 1)
-				active_attachable.activate_attachment(src, null, TRUE)
-				reload(user,MG)
-				return
-		active_attachable.reload_attachment(I, user)
-
-	else if(istype(I,/obj/item/cell/lasgun))
-		if(check_inactive_hand(user))
-			reload(user,I)
+	else if(istype(I, /obj/item/cell/lasgun) && check_inactive_hand(user))
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/energy.dm]: as above, if you reload an attachment with a lasgun cell wont this also trigger or potentially try to reload with a qdelled cell if the attachment qdels the cell as part of the reload

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm (hunk 1 above)
Comment (DominikPanic) [reply code/modules/projectiles/updated_projectiles/gun_helpers.dm]: Only one else-if will trigger.

Comment (Rohesie) [reply code/modules/projectiles/updated_projectiles/gun_helpers.dm]: A `return` would add clarity to the code.
This might be subjective, though.

**Pull request merged**